### PR TITLE
Fix OOM slice removal race

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
If the slice is already removed then we mostly encounter two different errors:

- `get next line: No such device (os error 19)`
- `open memory events file: /sys/fs/cgroup/test.slice/crio-$ID.scope/memory.events: No such file or directory (os error 2)`

To avoid such a race we now check after the errors if the file still exists. If not, then we assume an OOM.

#### Which issue(s) this PR fixes:

Fixes https://github.com/cri-o/cri-o/issues/8411

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Fixed OOM slice removal race to not report an out of memory event.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@haircommander @rphillips PTAL

File: conmon-rs/server/src/oom_watcher.rs [hunk 1]
```diff
@@ -252,8 +253,19 @@ impl OOMWatcher {
                                     };
                                 }
                                 Err(e) => {
-                                    error!("Checking for OOM failed: {}", e);
-                                    match tx.try_send(OOMEvent{ oom: false }) {
+                                    error!("Checking for OOM failed: {:#}", e);
+
+                                    let mut oom = false;
+                                    if !memory_events_file_path.exists() {
```
Comment (rphillips) [inline conmon-rs/server/src/oom_watcher.rs:259]: I'm not sure which exists() this is... A better call might be tokio's async call (untested):

```suggestion
 !tokio::fs::metadata("/some/file/path.txt").await?.is_ok()
```

Comment (saschagrunert) [reply conmon-rs/server/src/oom_watcher.rs]: Yep, done :+1:

**Comment by rphillips** [comment]
Nice!
/lgtm

**Pull request merged**